src/math/exp_fcn.c

```diff
@@ -0,0 +1,217 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+ *
+ */
+
+/* Include Files */
+#include <sof/math/exp_fcn.h>
+#include <sof/math/numbers.h>
+#include <sof/common.h>
+#include <rtos/bit.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#define BIT_MASK_Q62P2 0x4000000000000000LL
+#define CONVERG_ERROR 28823037607936LL	// error smaller than 1e-4,1/2 ^ -44.7122876200884
+#define BIT_MASK_LOW_Q27P5 0x8000000
+#define QUOTIENT_SCALE BIT(30)
+#define TERMS_Q23P9 8388608
+#define LSHIFT_BITS 8192
+
+/* inv multiplication lookup table */
+/* LUT = ceil(1/factorial(b_n) * 2 ^ 63) */
+const int64_t exp_iv_ilookup[19] = {
```

Comment:
I assume this needs to be static ?

Reply:
@lgirdwood no, this will be shared with the next pull request.